src/components/TodosFilter.tsx

```diff
@@ -0,0 +1,80 @@
+import classNames from 'classnames';
+import React, { useContext } from 'react';
+import { TodosContext } from './TodosContext';
+
+export const TodosFilter: React.FC = React.memo(() => {
+  const { todos, setTodos } = useContext(TodosContext);
+  const { filterBy, setFilterBy } = useContext(TodosContext);
+
+  const counter = () => todos.filter(todo => !todo.completed).length;
+  const counterCompleted = () => todos.filter(todo => todo.completed).length;
+
+  const clearCompleted = () => {
+    setTodos(todos.filter(todo => !todo.completed));
+  };
+
+  const filterHandler = (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+    setFilterBy(event.currentTarget.textContent || 'All');
+  };
+
+  return (
+    <>
+      {todos.length > 0 && (
+        <footer className="footer">
+          <span
+            className="todo-count"
+            data-cy="todosCounter"
+          >
+            {`${counter()} `}
+            {counter() > 1 ? 'items left' : 'item left'}
```

Comment:
pass it into variable and then use here (+ make all logic in your counter function - return ready string)